README.md

```diff
@@ -468,6 +469,170 @@ ANSWERS.addComponent('DirectAnswer', {
+    {{#if @root/eventType}}data-eventtype="{{@root/eventType}}"{{/if}}
+    {{#if @root/eventOptions}}data-eventoptions='{{{ json @root/eventOptions }}}'{{/if}}>
+    {{{displayText}}}
+  </a>
+  {{/inline}}
+```
+
+This is an example of a possible custom Direct Answer card. It has custom rendering that depends on
+the fieldType of the directAnswer, as well as a a custom analytics event.
+
+```js
+  class CustomDirectAnswerClassName extends ANSWERS.Component {
+    constructor(config, systemConfig) {
+      // If you need to override the constructor, make sure to call super(config, systemConfig) first.
+      super(config, systemConfig);
+    }
+
+    /**
+     * The setState method allows you to pass variables directly into your template.
+     * Here, data is the directAnswer data from the query.
+     * Below, we pass through a custom direct answers value, customValue.
+     * @param {Object} data
+     * @returns {Object}
+     */ 
+    setState(data) {
+      const { type, answer, relatedItem } = data;
+      this.associatedEntityId = data.relatedItem && data.relatedItem.data && data.relatedItem.data.id;
+      this.verticalConfigId = data.relatedItem && data.relatedItem.verticalConfigId;
+      return super.setState({
+        ...data,
+        customValue: this.getCustomValue(answer),
+        eventType: 'CUSTOM_EVENT',
+        eventOptions: {
+          searcher: "UNIVERSAL",
+          verticalConfigId: this.verticalConfigId,
+          entityId: this.associatedEntityId,
+        }
+      });
+    }
+
+    /**
+     * Formats a directAnswer answer based on its fieldType.
+     * @param {Object} answer the answer property in the directAnswer model
+     * @returns {string}
+     */ 
+    formatValue(answer) {
+      const { fieldType, value } = answer;
+      switch (fieldType) {
+        case "phone":
+          return {
+              url: 'http://myCustomWebsite.com/?mainPhone=' + value,
+              displayText: value,
+            };
+        case "rich_text":
+          return ANSWERS.formatRichText(value);
+        case "single_line_text":
+        case "multi_line_text":
+        default:
+          return value;
+      }
+    }
+
+    /**
+     * Computes a custom direct answer. If answer.value is an array, this method
+     * formats every value in the array and returns it, otherwise it just formats the single
+     * given value.
+     * @param {Object} answer
+     * @returns {Array<string>}
+     */ 
+    getCustomValue(answer) {
+      if (Array.isArray(answer.value)) {
+        return answer.value.map(value => this.formatValue(answer))
+      } else {
+        return [ this.formatValue(answer) ];
+      }
+    }
+
+    /**
+     * The name of your custom direct answer card. THIS is the value you will use in any config,
+     * such as defaultCard, when you want to specify this custom Direct Answer card.
+     * @returns {string}
+     */
+    static get type() {
+      return 'MyCustomDirectAnswerCard';
+    }
+  }
+
```

Comment:
I think custom components also need the 

```
  static areDuplicateNamesAllowed() {...}
```

func, is that right?

Reply:
the built in DirectAnswer doesn't currently set this (which defaults to false) so I think it should be fine.

Reply:
oh nice ok sgtm!